lib/response.js

```diff
@@ -56,8 +56,14 @@ var charsetRegExp = /;\s*charset\s*=/;
  * @public
  */
 
-res.status = function status(code) {
-  this.statusCode = code;
+res.status = function status(statusCode) {
+  // check that status code is valid
+  statusCode = parseInt(statusCode, 10);
```

Comment:
Basically, if the purpose of this isn't to catch obviously invalid input up front, then there doesn't seem a point to the PR to me, as we can just let Node.js catch it as it does today.